os/board/rtl8721csm/src/component/soc/realtek/amebad/fwlib/ram_hp/rtl8721dhp_app_start.c

```diff
@@ -52,9 +52,14 @@ extern unsigned int __PsramStackLimit;
 #define HEAP_LIMIT ((uintptr_t)&__StackLimit)
 #define PSRAM_HEAP_BASE ((uintptr_t)&__psram_bss_end__[0])
 #define PSRAM_HEAP_LIMIT ((uintptr_t)&__PsramStackLimit)
+#define SRAM_EBSS_SECTION SECTION("._ebss.data")
+
+SRAM_EBSS_SECTION
+static unsigned char srHeap[(CONFIG_SRAM_HEAP_SIZE*1024)+CONFIG_IDLETHREAD_STACKSIZE];
+
 void os_heap_init(void){
-	kregionx_start[0] = (void *)HEAP_BASE;
-	kregionx_size[0] = (size_t)(HEAP_LIMIT - HEAP_BASE);
+	kregionx_start[0] = (void *)(srHeap + CONFIG_IDLETHREAD_STACKSIZE);
+	kregionx_size[0] = (size_t)(CONFIG_SRAM_HEAP_SIZE*1024);
```

Comment:
It is sufficient to set the required size using CONFIG_SRAM_HEAP_SIZE*1024. This will guarantee that the kernel heap is 109KB.
However, what if 109KB space is not available in the SRAM ? We need to check using HEAP_LIMIT this and if less than 109KB is available, then the size needs to be set accordingly.